File: component/CommonComponent/src/main/java/com/energy/commoncomponent/bean/RotateDegree.java
package com.energy.commoncomponent.bean;

/**
 * Created by fengjibo on 2023/7/4.
 */
public enum RotateDegree {

    DEGREE_0(0),
    //
    DEGREE_90(1),
    //
    DEGREE_180(2),
    //
    DEGREE_270(3);

    private final int value;

    RotateDegree(int value) {
        this.value = value;
    }

    public int getValue() {
        return value;
    }

    public static RotateDegree valueOf(int value) {
        RotateDegree[] types = RotateDegree.values();
        for(RotateDegree type: types){
            if (type.value == value) {
                return type;
            }
        }
        return DEGREE_0;
    }
}

File: component/shared/src/main/java/com/mpdc4gsr/component/shared/common/RotateDegree.kt
package com.mpdc4gsr.component.shared.common

enum class RotateDegree(
    private val value: Int,
) {
    DEGREE_0(0),
    DEGREE_90(90),
    DEGREE_180(180),
    DEGREE_270(270),
    ;

    fun getValue(): Int = value
}


